shell/platform/android/io/flutter/view/FlutterMain.java

```diff
@@ -271,22 +281,41 @@ public void run() {
      */
     private static void initConfig(Context applicationContext) {
         try {
-            Bundle metadata = applicationContext.getPackageManager().getApplicationInfo(
-                applicationContext.getPackageName(), PackageManager.GET_META_DATA).metaData;
+            Bundle metadata = applicationContext
+                .getPackageManager()
+                .getApplicationInfo(applicationContext
+                    .getPackageName(), PackageManager.GET_META_DATA)
+                .metaData;
+
             if (metadata != null) {
                 sAotSharedLibraryPath = metadata.getString(PUBLIC_AOT_AOT_SHARED_LIBRARY_PATH, DEFAULT_AOT_SHARED_LIBRARY_PATH);
-                sAotVmSnapshotData = metadata.getString(PUBLIC_AOT_VM_SNAPSHOT_DATA_KEY, DEFAULT_AOT_VM_SNAPSHOT_DATA);
-                sAotVmSnapshotInstr = metadata.getString(PUBLIC_AOT_VM_SNAPSHOT_INSTR_KEY, DEFAULT_AOT_VM_SNAPSHOT_INSTR);
-                sAotIsolateSnapshotData = metadata.getString(PUBLIC_AOT_ISOLATE_SNAPSHOT_DATA_KEY, DEFAULT_AOT_ISOLATE_SNAPSHOT_DATA);
-                sAotIsolateSnapshotInstr = metadata.getString(PUBLIC_AOT_ISOLATE_SNAPSHOT_INSTR_KEY, DEFAULT_AOT_ISOLATE_SNAPSHOT_INSTR);
                 sFlutterAssetsDir = metadata.getString(PUBLIC_FLUTTER_ASSETS_DIR_KEY, DEFAULT_FLUTTER_ASSETS_DIR);
+                aotAbiSplit = metadata.getBoolean(PUBLIC_AOT_ABI_SPLIT_KEY, DEFAULT_AOT_ABI_SPLIT);
+
+                if (aotAbiSplit) {
```

Comment:
The `metadata.getString` lookups are intended to let applications override the filenames used to package the AOT assets in the APK.

I'd remove this block and instead change the default value for these lookups based on whether we're in `aotAbiSplit` mode.

Reply:
I wasn't sure about it since based on this comment https://github.com/flutter/engine/pull/8979#discussion_r284493367, we have a circular dependency: The default name depends on whether the snapshots are in `lib/`.  For now, I'm getting the value and wrapping it in a lib_*.so format.